### PR TITLE
report errors to github comment

### PR DESCRIPTION
**Pull request by lbergelson**

### CODE REVIEW

**Comment by jamesemery** [comment]
Where is the comment from the travis gods on this PR?

**Comment by lbergelson** [comment]
It didn't work because python failed to install correctly.

**Comment by lbergelson** [comment]
@jamesemery It's aaaalive!

**Comment by lbergelson** [comment]
@droazen Behold the beauty of this travis bot...

**Comment by jamesemery** [comment]
First comment, I would like to see what happens when there are 2 concurrent failing commits being run by travis.

**Comment by lbergelson** [comment]
@jamesemery I reverted the two breaking changes in 2 separate commits so we should see a pair of comments coming down the line.

File: .travis.yml [hunk 1]
```diff
@@ -205,16 +204,23 @@ after_success:
     git tag master;
     ./gradlew uploadArchives; 
   fi;
-
+- if [[ $TRAVIS_SECURE_ENV_VARS == true && $TRAVIS_EVENT_TYPE != cron ]]; then
+    if [[ ${TESTS_REQUIRE_GCLOUD} != "true" ]]; then
```
Comment (jamesemery) [inline .travis.yml]: We should just make this script do nothing if the authentication already exists. This feels like a point of confusion in the script that will cause problems if we change things later.

**Comment by lbergelson** [comment]
Er, actually, only 1 of them should fail because the one with both reversions wont.  I've reverted a revert...

**Comment by lbergelson** [comment]
and will revert that revert revert now to cause it to pass again...

**Comment by lbergelson** [comment]
hmn.. I might have a compile error by accident...
more reveerrts

File: .travis.yml [hunk 2]
```diff
@@ -205,16 +204,23 @@ after_success:
     git tag master;
     ./gradlew uploadArchives; 
   fi;
-
+- if [[ $TRAVIS_SECURE_ENV_VARS == true && $TRAVIS_EVENT_TYPE != cron ]]; then
```
Comment (jamesemery) [inline .travis.yml]: You duplicated all of this report uploading code in the after failure/after success block. Could that be condensed into the after script block and then the python script is run in the after failure block? I'm not sure what order these are all executed in...

File: resources_for_CI/Reporter.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
+import os
+import sys
+
+from github import Github
```
Comment (jamesemery) [inline resources_for_CI/Reporter.py]: I would add a comment either here or in a readme somewhere explaining exactly what this is doing regarding reading of comments/parsing comments and how it decides to update them.

File: .travis.yml (hunk 2 above)
Comment (lbergelson) [reply .travis.yml]: `after_script` happens **after** `after_failure` / `after_success` as far as I know so you can't do it nicely.

File: .travis.yml (hunk 1 above)
Comment (lbergelson) [reply .travis.yml]: That's a good idea.  It should be idempotent.

File: .travis.yml (hunk 2 above)
Comment (lbergelson) [reply .travis.yml]: I can set an environment variable here that tells it to run/notrun the python script later.  That's probably nicer.

File: .travis.yml (hunk 1 above)
Comment (lbergelson) [reply .travis.yml]: I think we should do this in a separate pr.

**Pull request merged**